db/data.py

```diff
@@ -124,7 +125,11 @@ def submit_low_level_data(mbid, data):
         )
 
     # The data looks good, lets see about saving it
-    write_low_level(mbid, data)
+    try:
+        write_low_level(mbid, data)
+    except sqlalchemy.exc.DataError as e:
+        raise db.exceptions.BadDataException(
+                "data is badly formed")
```

Comment:
I'm not sure what we should say here. I don't really want to expose the word `postgres` in this message